R/treeview.R

```diff
@@ -1,394 +1,463 @@
-			paste( sep = '\n' , 'Cluster branch mutations:',
+		ttdefmuts <- sapply(match(gtr1.1$data$cluster_id, sc0$cluster_id), function(isc0) {
+			if (is.na(isc0))
+				return("")
+			paste(sep = "\n", "Cluster branch mutations:",
 				gsub(
 					x = tryCatch(
 						stringr::str_wrap(
-							paste( collapse = ' ' , .sort.muts( cmuts[[  as.character( sc0$node_number[isc0] ) ]]$defining )  )
-							, width = 60
-						)
-						, error = function(e)browser() )
-					, patt = ' '
-					, rep = ', '
-				)
-				,'\n'
-			)
-		})
-		ttallmuts <- sapply( match( gtr1.1$data$cluster_id, sc0$cluster_id ) , function(isc0){
-			if ( is.na( isc0 ))
-				return( '' )
-			paste( sep = '\n' , 'All mutations:',
+							paste(collapse = " ", .sort.muts(cmuts[[as.character(sc0$node_number[isc0])]]$defining)),
+							width = 60),
+						error = function(e) browser()),
+					pattern = " ",
+					replacement = ", "),
+				"\n")
+		}) # end of sapply
+
+		ttallmuts <- sapply(match(gtr1.1$data$cluster_id, sc0$cluster_id), function(isc0) {
+			if (is.na(isc0))
+				return("")
+			paste(sep = "\n", "All mutations:",
 				gsub(
 					x = stringr::str_wrap(
-						paste( collapse = ' ' , .sort.muts(  cmuts[[  as.character( sc0$node_number[isc0] ) ]]$all ) )
-						, width = 60
-					)
-					, patt = ' '
-					, rep = ', '
-				)
-				,'\n'
-			)
-		})
-
-		gtr1.1$data$defmuts = ttdefmuts
-		gtr1.1$data$allmuts = ttallmuts
-		if ( !is.null( mut_regex  )){
+						paste(collapse = " ",
+						      .sort.muts(cmuts[[as.character(sc0$node_number[isc0])]]$all)),
+						width = 60),
+					pattern = " ",
+					replacement = ", "),
+				"\n")
+		}) # end of sapply
+
+		gtr1.1$data$defmuts <- ttdefmuts
+		gtr1.1$data$allmuts <- ttallmuts
+		if (!is.null(mut_regex)) {
 			#gtr1.1$data$label <- ''
-			for ( mre in mut_regex){
-				i <- which( grepl(gtr1.1$data$allmuts, patt = mre ) )
+			for (mre in mut_regex) {
+				i <- which(grepl(gtr1.1$data$allmuts, pattern = mre))
 				#gtr1.1$data$label[i] <- '*'
-				gtr1.1$data[[ mre ]] <-  grepl(gtr1.1$data$allmuts, patt = mre )
+				gtr1.1$data[[mre]] <-  grepl(gtr1.1$data$allmuts, pattern = mre)
 			}
 			#gtr1.1 <- gtr1.1 + geom_tiplab( size = 16, colour = 'red' )
 		}
 
-		genotype <- as.data.frame( gtr1.1$data[ gtr1.1$data$node <= Ntip(tr2) , c( 'label', mut_regex ) ] )
-		rownames( genotype ) <- genotype$label
-		genotype <- genotype[ ,-1, drop = FALSE]
+		genotype <- as.data.frame(gtr1.1$data[gtr1.1$data$node <= ape::Ntip(tr2),
+		                                      c("label", mut_regex)])
+		rownames(genotype) <- genotype$label
+		genotype <- genotype[, -1, drop = FALSE]
 
 		# make html widget
-		gtr1.1$data$mouseover = sapply( 1:length( ttdfs ), function(i){
-			 paste0(  'Statistics:\n',  ttdfs[i],   '\n\nGeography:\n', ttregtabs[i], '\n\nCo-circulating with:\n', ttcocirc[i], '\n\n', ttdefmuts[i],'\n', ttallmuts[i],'\n' , collapse = '\n')
+		gtr1.1$data$mouseover <- sapply(seq_len(length(ttdfs)), function(i) {
```

Comment:
seq_along is a little more concise in this setting.
```suggestion
		gtr1.1$data$mouseover <- sapply(seq_along(ttdfs), function(i) {
```

Reply:
Fixed.